src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,34 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herd) {
+    if (herd instanceof Herbivore && herd.hidden === false) {
+      herd.health -= 50;
+    }
+
+    Animal.alive.splice(Animal.alive.indexOf(herd), 1);
+  }
```

Comment:
```suggestion
  bite(herd) {
    if (herd instanceof Herbivore && herd.hidden === false) {
      herd.health -= 50;
    }

    Animal.alive.splice(Animal.alive.indexOf(herd), 1);
  }
```

it will delete animal on the first bite, when it's health is still 50, so you should add if statement to check if herd's health is equal to 0, only then you can delete herd from alive array